cmd/internal/kube/daemonset/kube_events_worker.go

```diff
@@ -16,151 +15,180 @@ var (
 
 // handlePodAddEvent handles the event when a pod is added to the Kubernetes cluster.
 // It performs the following steps:
-// 1. Retrieves the pod by its UID.
-// 4. Filters out pods that already have the agent sidecar container.
-// 5. Inspects the pod for required environment variables.
-// 6. Adds the pod arguments to a map and starts the API dump process.
-func (d *Daemonset) handlePodAddEvent(podUID types.UID) {
-	pods, err := d.KubeClient.GetPodsByUIDs([]types.UID{podUID})
-	if err != nil {
-		printer.Errorf("Failed to get pod for k8s added event, podUID: %s, error: %v\n", podUID, err)
-		return
-	}
-	if len(pods) == 0 {
-		printer.Infof("No pod found for k8s added event, podUID: %s\n", podUID)
-		return
-	}
-	if len(pods) > 1 {
-		printer.Errorf("Multiple pods found for single UID, this should not happen, podUID: %s\n", podUID)
-		return
-	}
+// 1. Check if the pod does not have the agent sidecar container.
+// 3. Adds the pod arguments to a map and change state to PodPending.
+func (d *Daemonset) handlePodAddEvent(pod coreV1.Pod) {
 
 	// Filter out pods that do not have the agent sidecar container
-	podsWithoutAgentSidecar, err := d.KubeClient.FilterPodsByContainerImage(pods, agentImage, true)
+	podsWithoutAgentSidecar, err := d.KubeClient.FilterPodsByContainerImage([]coreV1.Pod{pod}, agentImage, true)
 	if err != nil {
 		printer.Errorf("Failed to filter pod by container image: %v\n", err)
 		return
 	}
 	if len(podsWithoutAgentSidecar) == 0 {
-		printer.Infof("Pod already has agent sidecar container, skipping, podUID: %s\n", podUID)
+		printer.Infof("Pod already has agent sidecar container, skipping, podUID: %s\n", pod.UID)
 		return
 	}
 
-	pod := podsWithoutAgentSidecar[0]
-
-	args, err := d.inspectPodForEnvVars(pod)
-	if err != nil {
-		switch err {
-		case allRequiredEnvVarsAbsentErr:
-			printer.Debugf("None of the required env vars present, skipping pod: %s\n", pod.Name)
-		case requiredEnvVarMissingErr:
-			printer.Errorf("Required env var missing, skipping pod: %s\n", pod.Name)
-		default:
-			printer.Errorf("Failed to inspect pod for env vars, pod name: %s, error: %v\n", pod.Name, err)
-		}
-		return
+	// Empty podArgs object for PodPending state
+	args := &PodArgs{
+		PodName: pod.Name,
+		// though 1 buffer size is enough, keeping 2 for safety
+		StopChan: make(chan error, 2),
 	}
```

Comment:
This creation of a PodArgs in a particular state should be encapsulated in a function?  It appears at least twice.